pkg/kubelet/cm/deviceplugin/device_plugin_stub.go

```diff
@@ -70,7 +70,7 @@ func (m *Stub) Start() error {
 	// Wait till grpc server is ready.
 	for i := 0; i < 10; i++ {
 		services := m.server.GetServiceInfo()
-		if len(services) > 1 {
+		if len(services) > 0 {
```

Comment:
This whole loop is pretty much useless and is a "fancy sleep" that executes enough code so that the server goes up on the CI.
Not really worth changing it

Reply:
This `if` statement will never be got inside as `GetServiceInfo()` will return only one service, so if this is a "fancy sleep", maybe we can just sleep here without any conditionals.

Reply:
I already fixed the insanity in the stub device plugin in https://github.com/kubernetes/kubernetes/pull/52612

The sane way to do this is to send a blocking connection (with timeout) to the server after starting it.